src/FastRandom.cpp

```diff
@@ -13,109 +13,49 @@
 #elif defined (_MSC_VER)
 	#define ATTRIBUTE_TLS static __declspec(thread)
 #else
-	#error "Unknown thread local storage qualifier"
+	#define ATTRIBUTE_TLS thread_local
```

Comment:
This is wrong - you can't be sure that this is true for all other compilers. You should specify the attributes for known compilers and leave the `#else` branch intact.